source/transfer_list.rst

```diff
@@ -141,15 +141,39 @@ The TE header is defined in :numref:`tab_te_header`.
      - The size of this entry header in bytes. This field is set to 8 for the TE header layout described in this version of the table.
 
 
-Entry type ranges
+Entry type allocation
 -----------------
 
-The content of the data section is determined by the tag id. The tag id space contains two ranges:
-
- #. Standard range, and
- #. Non-standard range
-
-The *tag_id* ranges are described in :numref:`tab_tag_id_ranges`.
+Tag IDs must be allocated in this specification before use. A new tag ID can be
+allocated by submitting a pull request to this repository that adds a description
+of the respective TE data layout to this specification. Tag IDs do not have to be
+allocated in order. Submitters are encouraged to try to group tag IDs together in
+logical clusters (e.g. all tags related to a particular chipset or to a
+particular firmware project could use adjacent tag numbers), but there are no
+predefined ranges and no reservations of tag ranges for specific use. The
+initially legal range for tag allocations is {0, ..., 0xf_ffff} (more ranges
+will be opened up when necessary).
+
+Tags are expected to have a simple layout and each tag should only represent
+data for a single logical concept. Data for multiple distinct concepts should be
+split across different tags, even if they're always expected to appear together
+on the first platform adding the tag (to encourage reusability in different
+situations). A list of objects can be represented by adding multiple TEs with
+the same tag ID to the TL.
+
+The {0x10_0000, ... 0x10_ffff} range is reserved for non-standardized use. Anyone
```

Comment:
At some point we may need to expand the standard range and it would be nice to keep it contiguous. How about we put the non-standard range at the end and reserve the middle part instead?

Reply:
Yes, either the end or the beginning would make most sense. I think maybe putting it at the beginning would be a little better so that the high bits of the tag ID remain available a little longer for potential future special meanings?

Reply:
I meant something like in the ARM SMCCC where bits 31 and 30 of the FID have special meaning. Of course this would only make sense if we later come up with a use case for splitting all tag types into two separate groups like that (and at the point where we make that decision, all existing tags would have to fall into the same group), which I can't anticipate right now and don't think is likely. I was just trying to preserve options for the future as much as possible.

The idea was to declare something like 0x0 - 0xffff as non-standard and everything from 0x10000 upward as standardized, so the tags in the standardized range would still grow upwards, not downwards (just not starting at 0x0).

Reply:
I suggest using the lower range for tags and trying to bunch common ones at a the lower range <256 since this can help with code size when the literals can fit in a single instruction

Reply:
Moved non-standard range to the top of the tag space in latest push.